examples/features/metadata/client/main.go

```diff
@@ -0,0 +1,279 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"google.golang.org/grpc"
+	pb "google.golang.org/grpc/examples/features/proto/echo"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	address         = "localhost:9527"
+	timestampFormat = time.StampNano // "Jan _2 15:04:05.000"
+	streamingCount  = 10
+)
+
+func unaryCallWithMetadata(c pb.EchoClient, message string) {
+	fmt.Printf("--- unary ---\n")
+	// Create metadata and context.
+	md := metadata.Pairs("timestamp", time.Now().Format(timestampFormat))
+	ctx := metadata.NewOutgoingContext(context.Background(), md)
+
+	// Make RPC using the context with the metadata.
+	var header, trailer metadata.MD
+	r, err := c.UnaryEcho(ctx, &pb.EchoRequest{Message: message}, grpc.Header(&header), grpc.Trailer(&trailer))
+	if err != nil {
+		log.Fatalf("failed to call UnaryEcho: %v", err)
+	}
+
+	if t, ok := header["timestamp"]; ok {
+		fmt.Printf("timestamp from header:\n")
+		for i, e := range t {
+			fmt.Printf(" %d. %s\n", i, e)
+		}
+	}
```

Comment:
Either:

- `else { log.Fatal() }`, or

- remove the `if` and just assume it's present.  Actually this will work and not output anything if it's missing, since we `range` over `t`, and ranging over a `nil` slice is fine.

In the second case...nothing bad happens if things are broken and the metadata is not present.  Is that OK?

Reply:
Added `else`.